### PR TITLE
bunch of write barrier fix

### PR DESCRIPTION
**Pull request by leirocks**
these issues are found by running octane, -nonative passed on my machine with this change. there might be JIT related issue still under investigation

### CODE REVIEW

**Comment by leirocks** [comment]
@dotnet-bot test Linux tests please

**Comment by leirocks** [comment]
@dotnet-bot test Windows x86_debug please

**Comment by leirocks** [comment]
@dotnet-bot test Windows x86_debug please

File: lib/Runtime/Language/FunctionCodeGenRuntimeData.h [hunk 1]
```diff
@@ -36,8 +36,8 @@ namespace Js
     public:
         FunctionBody *GetFunctionBody() const;
         FunctionCodeGenRuntimeData *GetNext() const { return next; };
-        FunctionCodeGenRuntimeData **GetInlinees() const { return inlinees; }
-        FunctionCodeGenRuntimeData **GetLdFldInlinees() const { return ldFldInlinees; }
+        Field(FunctionCodeGenRuntimeData*)* GetInlinees() const { return (Field(FunctionCodeGenRuntimeData*)*)(void*)inlinees; }
```
Comment (jianchun) [inline lib/Runtime/Language/FunctionCodeGenRuntimeData.h]: >return (Field(FunctionCodeGenRuntimeData*)*)(void*)inlinees; [](start = 66, length = 60)

"return inlinees" should work? Same for "return ldFldInlinees".

File: lib/Runtime/Base/FunctionBody.cpp [hunk 2]
```diff
@@ -6294,7 +6298,7 @@ namespace Js
     {
         Assert(inlineCacheIndex < this->GetInlineCacheCount());
 
-        FunctionCodeGenRuntimeData ** data = (FunctionCodeGenRuntimeData **)this->GetCodeGenGetSetRuntimeDataWithLock();
+        auto data = (FunctionCodeGenRuntimeData **)this->GetCodeGenGetSetRuntimeDataWithLock();
```
Comment (jianchun) [inline lib/Runtime/Base/FunctionBody.cpp]: >(FunctionCodeGenRuntimeData **) [](start = 20, length = 31)

Cast "(FunctionCodeGenRuntimeData **)" is no longer needed?

**Comment by leirocks** [comment]
@dotnet-bot test Linux tests please

**Pull request merged**